packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,517 @@
+/// layouts for many different screen sizes. It separates the screen into
+/// several sections, also referred to as "slots". The Widget defines slots for
+/// the different placements of navigational elements and two slots for a body
+/// and secondaryBody. All of these slots take a [SlotLayout] Widget. This class
+/// works in junction with [SlotLayout] and [SlotLayoutConfig]. This class
+/// handles the placement of the slots on the screen and animations regarding
+/// their macromovements.
+///
+/// See also:
+///
+///  * [SlotLayout], which handles the actual switching and animations between
+/// elements based on [Breakpoint]s.
+///  * [SlotLayoutConfig], which holds information regarding the actual Widgets
+/// and the desired way to animate between switches. Often used within
+/// [SlotLayout].
+///  * [Design Doc](https://flutter.dev/go/adaptive-layout-foldables).
+///  * [Material Design 3 Specifications] (https://m3.material.io/foundations/adaptive-design/overview).
+class AdaptiveLayout extends StatefulWidget {
+  /// Creates an [AdaptiveLayout] widget.
+  const AdaptiveLayout({
+    this.primaryNavigation,
+    this.secondaryNavigation,
+    this.topNavigation,
+    this.bottomNavigation,
+    this.body,
+    this.secondaryBody,
+    this.bodyRatio,
+    this.internalAnimations = true,
+    this.bodyOrientation = Axis.horizontal,
+    super.key,
+  });
+
+  /// The slot placed on the beginning side of the screen; meaning on the left
+  /// when the textDirection is LTR and on the right when it is RTL.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? primaryNavigation;
+
+  /// The slot placed on the end side of the screen; meaning on the right
+  /// when the textDirection is LTR and on the left when it is RTL.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? secondaryNavigation;
+
+  /// The slot placed on the top part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? topNavigation;
+
+  /// The slot placed on the bottom part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? bottomNavigation;
+
+  /// The slot that fills the rest of the space in the center.
+  final SlotLayout? body;
+
+  /// A supporting slot for [body]. Has a sliding entrance animation by default.
+  /// The default ratio for the split between [body] and [secondaryBody] is so
+  /// that the split axis is in the center of the screen.
+  final SlotLayout? secondaryBody;
+
+  /// Defines the fractional ratio of [body] to the [secondaryBody].
```

Comment:
I haven't read what SlotLayout/SlotConfiguration allow one to specify yet. It seems odd to have this parameter (and bodyOrientation, internalAnimations) given all of the abstractions (breakpoints too) we've defined to specify layouts.

Reply:
The bodyRatio parameter is there to support cases where the developer places widgets of fluid size into body and secondaryBody and wants to define their sizes relationally. The body animation with the slide in only works with this kind of configuration because you can't assess the body and secondaryBody's sizes while you're laying them out. With the navigational slots this isn't necessary because the sizes are inflexible but most of the time the body size can't be hardcoded.